src/QMCDrivers/WFOpt/QMCCostFunction.cpp

```diff
@@ -168,7 +168,13 @@ void QMCCostFunction::GradCost(std::vector<Return_rt>& PGradient,
 
 void QMCCostFunction::getConfigurations(const std::string& aroot)
 {
-  //makeClones(W,Psi,H);
+  // Set all the myVars indices in the wavefunction clones.
+  // If not set, the parameter derivative pieces in evaluateDerivatives
+  // end up in the wrong indices in dlogpsi and dhpsioverpsi.
+  for (int i = 0; i < psiClones.size(); i++)
+    psiClones[i]->checkOutVariables(OptVariablesForPsi);
```

Comment:
Trying to verify this is a good way to do this I fell right into having to reason about `QMCostFunction` `CloneManager` and `psiClones` multiplicity, in the amount of time I have I failed.
The naming is perfectly backwards for OptVariablesForPsi. I think there is a psiClones per "thread." I think they are just getting indexing set up and this isn't a checkout in the sense its used elsewhere in the code where psiClones now is using a memory resource or the like from OptVariableForPsi?

We're commited to just deleting this code ASAP correct?
Based on QMCCostFunctionBatched I assume so.

Reply:
checkOutVariables is called on Psi in QMCCostFunctionBase::put.  I don't know why that isn't sufficient to set the indices before makeClones is called in VMC::run.  I'll have to look into it.

Reply:
The DiracDeterminant constructor calls buildOptVariables.  RotatedSPO::buildOptVariables calls myVars.clear().  A better solution might be to make sure buildOptVariables only initializes the parameters once, rather than every time a clone is made.

Reply:
It appears that if I protect the body of RotatedSPOs::buildOptVariables against reinitialization, some of the other changes in this PR aren't necessary for the test to pass.